deeptime/plots/implied_timescales.py

```diff
@@ -0,0 +1,192 @@
+            self._its_stats = np.asarray(its_stats).transpose(0, 2, 1)
+            if not (self.its_stats.ndim == 3 and self.its_stats.shape[0] == self.n_lagtimes and
+                    self.its_stats.shape[1] == self.n_processes):
+                raise ValueError(f"its_stats should be of shape (lagtimes={self.n_lagtimes}, "
+                                 f"processes={self.n_processes}, samples={self.n_processes}) but was "
+                                 f"{self.its_stats.shape}")
+        else:
+            self._its_stats = None
+        ix = np.argsort(self.lagtimes)
+        self._lagtimes = self._lagtimes[ix]
+        self._its = self._its[ix]
+        self._its_stats = None if self._its_stats is None else self._its_stats[ix]
+
+    @property
+    def lagtimes(self) -> np.ndarray:
+        r""" Yields the lagtimes corresponding to an instance of this class. """
+        return self._lagtimes
+
+    @property
+    def its(self) -> np.ndarray:
+        r""" An ndarray of shape (`n_lagtimes`, `n_processes`) containing the timescales of each process. """
+        return self._its
+
+    @property
+    def its_stats(self) -> Optional[np.ndarray]:
+        r""" An ndarray of shape (`n_lagtimes`, `n_processes`, `n_samples`) or representing the timescales
+        of each process in each sample or `None` if no samples are available. """
+        return self._its_stats
+
+    @property
+    def n_lagtimes(self) -> int:
+        r""" Number of lagtimes. """
+        return len(self.lagtimes)
+
+    @property
+    def n_processes(self) -> int:
+        r""" Number of processes. """
+        return self.its.shape[1]
+
+    @property
+    def n_samples(self) -> int:
+        r""" Number of samples. """
+        return 0 if self.its_stats is None else self.its_stats.shape[2]
+
+    @staticmethod
+    def from_models(models, n_its=None):
+        r""" Converts a list of models to a :class:`ImpliedTimescalesData` object.
+
+        Parameters
+        ----------
+        data : list of models
+            The input data. Models with and without samples to compute confidences should not be mixed.
+        n_its : int or None, optional
+            Number of timescales to compute.
+
+        Returns
+        -------
+        its_data : ImpliedTimescalesData
+            The data object.
+        """
+        if not isinstance(models, (list, tuple)):
+            models = [models]
+
+        if len(models) == 0:
+            raise ValueError("Data cannot be empty.")
+        assert all(callable(getattr(model, 'timescales', None)) for model in models), \
+            "all models need to have a timescales method"
+        assert all(hasattr(model, 'lagtime') for model in models), "all models need a lagtime attribute or property"
+
+        is_bayesian = hasattr(models[0], 'prior') and hasattr(models[0], 'samples')
+        lagtimes = []
+        its = []
+        its_stats = [] if is_bayesian else None
+
+        for model in models:
```

Comment:
One could check for missing processes and pad arrays here.

Reply:
added padding in the ctor call